runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -275,14 +278,35 @@ public void write(IToken token, Object data) throws OverwriteException {
 
     /**
      * Read the given object from an address and streams.
-     *
+     * It is a wrapper of readNative to handle InterruptedException.
      * @param address An address to read from.
      * @param options Read options for this particular write (i.e. configure caching behavior)
      * @return A result, which be cached.
      */
     public @Nonnull
     ILogData read(long address, @NonNull ReadOptions options) {
+        int count = 0;
+        while (true) {
+            try {
+                return readNative (address, options);
+            } catch (InterruptedException e) {
+                if (++count == MAX_RETRIES) {
+                    log.error ("It has retried {} times, throw an exception {}", MAX_RETRIES, e);
```

Comment:
The last two braces should be deleted otherwise e.toString() would fill the replace holder.